tests/test_stepwise_reward_trainer.py

```diff
@@ -0,0 +1,178 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tempfile
+import unittest
+
+import torch
+from datasets import Dataset, load_dataset
+from transformers import AutoModelForTokenClassification, AutoTokenizer, EvalPrediction
+from transformers.testing_utils import require_peft
+from transformers.utils import is_peft_available
+
+from trl import StepwiseRewardTrainer, StepwiseRewardConfig, maybe_apply_chat_template
+from trl.trainer import compute_accuracy
+from trl.trainer.stepwise_reward_trainer import _tokenize
+
+
+if is_peft_available():
+    from peft import LoraConfig, TaskType
+
+
+class StepwiseRewardTrainerTester(unittest.TestCase):
+    def setUp(self):
+        self.model_id = "hf-internal-testing/tiny-random-LlamaForCausalLM"
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_id)
+        self.tokenizer.chat_template = "{% for message in messages %}{{'<|im_start|>' + message['role'] + '\n' + message['content'] + '<|im_end|>' + '\n'}}{% endfor %}{% if add_generation_prompt %}{{ '<|im_start|>assistant\n' }}{% endif %}"
+        self.model = AutoModelForTokenClassification.from_pretrained(self.model_id, num_labels=2)
+
+    def test_token_level_accuracy(self):
+        dummy_eval_predictions = EvalPrediction(
+            torch.FloatTensor([[[0.1, 0.9], [0.1, 0.9]], [[0.1, 0.9], [0.9, 0.1]]]),
+            torch.LongTensor([[-100, 1], [-100, 1]]),
+        )
+        accuracy = compute_accuracy(dummy_eval_predictions)
+        self.assertEqual(accuracy["accuracy"], 0.5)
+
+    def test_preprocessing_conversational(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            dummy_dataset = load_dataset(
+                "trl-internal-testing/zen", "conversational_stepwise_preference", split="train"
+            )
+            training_args = StepwiseRewardConfig(output_dir=tmp_dir, report_to="none", max_length=512)
+            trainer = StepwiseRewardTrainer(
+                model=self.model,
+                args=training_args,
+                tokenizer=self.tokenizer,
+                train_dataset=dummy_dataset,
+            )
+            dummy_dataset = dummy_dataset.map(maybe_apply_chat_template, fn_kwargs={"tokenizer": self.tokenizer})
+            dummy_dataset = dummy_dataset.map(
+                _tokenize,
+                batched=True,
+                fn_kwargs={"tokenizer": self.tokenizer, "max_length": 512, "post_step_separator": "\n"},
+            )
+            self.assertDictEqual(trainer.train_dataset[:], dummy_dataset[:])
+
+    def test_preprocessing_standard(self):
```

Comment:
I think this test can be merged with the previous with `    @parameterized.expand([("conversational_stepwise_preference",), ("standard_stepwise_preference",)])`

Reply:
Indeed